pymeasure/instruments/tektronix/tek371A.py

```diff
@@ -828,32 +716,340 @@ def wait_for_srq(self):
             time.sleep(0.1)
         self.srq_called = False
 
-    def activate_srq(self):
+    def discard_and_disable_all_events(self):
+        """
+        Discard all events (empty the all events in queue) and disable the notification for
+        all events via all mechanism.
+        :return: None
+        """
+        self.adapter.connection.discard_events(pyvisa_constants.VI_ALL_ENABLED_EVENTS,
+                                               pyvisa_constants.EventMechanism.all)
+        self.adapter.connection.disable_event(
+            pyvisa_constants.VI_ALL_ENABLED_EVENTS,
+            pyvisa_constants.EventMechanism.all)
+
+    def enable_srq_event(self):
         """
         Config the instrument for srq assertion on operation complete
         and creates a function for handling the srq event.
         :return: None
         """
 
         def event_handler(resource, event, user_handle):
+            log.info("Handled event %s on %s", event.event_type, resource)
+            log.info("User_Handle %s", user_handle)
             self.srq_called = True
-            # print(f"Handled event {event.event_type} on {resource}")
-            # self.adapter.connection.disable_event(event.event_type, event.event_mech)
-            # self.adapter.connection.uninstall_handler(event_type, wrapped, user_handle)
 
         event_type = pyvisa_constants.VI_EVENT_SERVICE_REQ
         event_mech = pyvisa_constants.EventMechanism.handler
         wrapped = self.adapter.connection.wrap_handler(event_handler)
-        user_handle = self.adapter.connection.install_handler(event_type, wrapped, 42)
+        self.adapter.connection.install_handler(event_type, wrapped, None)
         self.adapter.connection.enable_event(event_type, event_mech, None)
         self.srq = True
         self.opc = True
 
-    #########################################################################
-    # #
-    #########################################################################
-
     def initialize(self):
         """ Initialize the instrument. Settings are the same as at power-up"""
         log.info("Initializing the instrument.")
         self.write("INIt")
+        self.discard_and_disable_all_events()
+        time.sleep(1)
+
+    #########################################################################
+    #  AUXILIARY CLASSES
+    #########################################################################
+
+    class Tek371AEvent(IntFlag):
+        """
+        Auxiliary class create for translating the instrument 8 bits_status_string into
+        an Enum_IntFlag that will help to the user to understand such status.
+        """
+        # Event code from the instrument has to be interpreted as follows:
+        #
+        # response --> 'EVENT code'
+
+        #  SYSTEM EVENT CODES
+        NO_ERROR = 0
+        POWER_ON = 401
+        MEASURE_SIGNAL_OPERATION_COMPLETE = 402
+        USER_REQUEST = 403
```

Comment:
I think an IntEnum is more appropriate here.
IntFlag supposes, that you can have several flags at the same time and the total value is the sum of the individual values. A typical example are status bytes, where each bit signifies something different. A value of 5, for example, would indicate the first (value of 1) and the third flag (value of 4).

Reply:
Done. Thanks